test/wdio/slot-hide-content/cmp.test.tsx

```diff
@@ -1,31 +1,52 @@
-import { setupDomTests, waitForChanges } from '../util';
+import { Fragment, h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
 
 /**
  * Tests for projected content to be hidden in a `scoped` component
  * when it has no destination slot.
  */
 describe('slot-hide-content', function () {
-  const { setupDom, tearDownDom } = setupDomTests(document);
-  let app: HTMLElement | undefined;
-  let host: HTMLElement | undefined;
+  const clickHandler = () => {
+    document.querySelectorAll('.test-cmp').forEach((ref) => ref.setAttribute('enabled', true));
+  };
 
   beforeEach(async () => {
-    app = await setupDom('/slot-hide-content/index.html');
+    render({
+      template: () => (
+        <>
+          <slot-hide-content-scoped className="test-cmp">
+            <p id="slotted-1">Hello</p>
+          </slot-hide-content-scoped>
+
+          <slot-hide-content-open className="test-cmp">
+            <p id="slotted-2">Hello</p>
+          </slot-hide-content-open>
+
+          <button type="button">Enable slot</button>
+        </>
+      ),
+    });
+
+    await $('button').waitForExist();
+
+    document.querySelector('button').addEventListener('click', clickHandler);
   });
 
-  afterEach(tearDownDom);
+  afterEach(() => {
+    document.removeEventListener('click', clickHandler);
+  });
```

Comment:
You can remove the need for this if you bind to a function in the `beforeEach` rather than querying and adding the event listener